internal/v2/fake_s2av2/fake_s2av2_test.go

```diff
@@ -0,0 +1,113 @@
+package fake_s2av2
+
+import (
+	"testing"
+	s2av2pb "github.com/google/s2a-go/internal/proto/v2/s2a_go_proto"
+	commonpb "github.com/google/s2a-go/internal/proto/v2/common_go_proto"
+)
+
```

Comment:
Should we have this test suite actually setup, run and tear down the service? Is there a preferred Go package for this? I found the net/http package, it looks like it's for testing web applications, could we use it here?

Reply:
Ya I think we should change up how the unit tests are structure here (sorry in advance if this is painful to refactor - it will be annoying in this PR but the follow-ups for the other RPCs will be easy because we'll do the setup here).

Step #1: In the test file, create the following function:
```
func startFakeS2Av2Server() (address string, stop func(), err error) {
  // Pick an unused port, set address to "localhost:" + that port
  s := grpc.NewServer()
  // Register your fake s2av2 implementation with this server
  go func() {
     if err := s.Serve(listener); err != nil {
        // log error
      }
   }
  return address, func() { s.Stop() }, nil
}
``

Step #2: Build a table test with 3 fields in the struct: description, request, expectedResponse. The body of the test would go something like this:

```
address, stop, err := startFakeS2Av2Server()
// Might need to use a wait group or similar to make sure there is no flakiness
// if the server is slop to start up

// Create a stream to the server, send the request, get the response, check that the response is the expected one, and close the stream

stop()
```

Step #3: How to efficiently compare protos (apologies I didn't think to mention this yesterday when we were talking): we should use the `cmp.Diff` and `protocmp.Transform()` APIs, e.g. as is done here:

https://github.com/google/s2a-go/blob/e978c1436cecf397b4c1178c7e56519941b7d196/internal/handshaker/handshaker_test.go#L1138

Happy to have a meeting to discuss any of this since the details are a bit complicated. :)

Reply:
Thanks so much for this explanation!! It was super helpful :)